### DISCUSSION TITLE
Migrating to ParseSwiftOG: From parse-community/Parse-Swift

### DISCUSSION THREAD

**Discussion by cbaker6** [netreconlab/Parse-Swift#70] [General]
For anyone who has built their apps or frameworks with the parse-community Parse-Swift SDK, it is recommended to point your SPM packages to https://github.com/netreconlab/Parse-Swift/ instead. **I estimate Parse-Swift<sup>OG</sup> to be 1+ years ahead of `parse-community/Parse-Swift` in development time.** Assuming you were on version [4.14.2](https://github.com/parse-community/Parse-Swift/releases/tag/4.14.2) or below from the parse-community repo, transitioning is straight-forward and non-breaking as all you need to: 

1. Open your project, click on your project in the `Project Navigator` panel
1. Select the name of your project under Project in the right panel
1. Click the `Package Dependencies` tab 
1. Remove the `parse-comunnity` version of `Parse-Swift` from your project
1. Add the https://github.com/netreconlab/Parse-Swift/ and for `Dependency Rule`, select `Up to Next Major Version` and input `4.16.2` for the specific version and `< 5.0.0`

**You will automatically gain all of the feature additions and bug fixes detailed [here](https://github.com/netreconlab/Parse-Swift/compare/4.15.0...4.16.2) without having to make any changes to your code.**

## Migrating to 5.0.0+
Preparing to migrate to Parse-Swift 5.0.0+ will require some changes to your code as it introduces a number of breaking changes (see #74 to learn how to migrate). Though some of the changes will feel different at first, they will lead to easier development in the long-run as all synchronous networking and storage calls have been removed as they are problematic for an asynchronous SDK such as Parse-Swift.

**Comment by RhiannonPedroncelliWE** [comment]
It's great to see that the Parse-Swift SDK is still being actively maintained and improved. Transitioning to the new repository seems like a straightforward process for those who were using version 4.14.2 or below. However, it's important to note that migrating to version 5.0.0+ will require some changes to code due to breaking changes, but it will ultimately lead to easier development. It's always a good idea to keep up-to-date with the latest version of any SDK or framework to take advantage of new features and bug fixes.

**Comment by GailZelkindmg** [comment]
Thank you for sharing this information! It's great to know that transitioning from the parse-community Parse-Swift SDK to the netreconlab/Parse-Swift SDK is straightforward and can be done without making any changes to the code. This will surely help developers who are currently using the parse-community Parse-Swift SDK and want to take advantage of the feature additions and bug fixes provided by the netreconlab/Parse-Swift SDK.
